sdb/commands/zfs/btree.py

```diff
@@ -0,0 +1,56 @@
+#
+# Copyright 2019 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pylint: disable=missing-docstring
+
+from typing import Iterable
+
+import drgn
+import sdb
+
+
+class Btree(sdb.Walker):
+    """ walk  btree """
+
+    names = ["zfs_btree"]
+    input_type = "zfs_btree_t *"
+
+    def _val(self, start, size, idx):
+        return start + (size * idx)
+
+    def _helper(self, node: drgn.Object, elem_size : int) -> Iterable[drgn.Object]:
```

Comment:
since `elem_size` is constant for this invocation, and it's used in all the helper functions, would it make sense to store it in an instance variable, rather than passing it around everywhere?

Reply:
Yeah - I think that makes sense. I added it as an instance variable, let me know if you think I can improve the style - my python's pretty rusty.